src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -517,9 +517,9 @@ public static XDocument ReadExisting(string path, ILogger log)
                     props.AddRange(GenerateGroupsWithConditions(buildCrossPropsGroup, isMultiTargeting, CrossTargetingCondition));
                 }
 
-                // ContentFiles are read by the build task, not by NuGet
-                // for UAP with project.json.
-                if (request.ProjectStyle != ProjectStyle.ProjectJson)
+                // Write out contentFiles only for XPlat PackageReference projects.
+                if (request.ProjectStyle != ProjectStyle.ProjectJson
+                    && request.Project.RestoreMetadata?.SkipContentFileWrite != true)
```

Comment:
Is simply checking `SkipContentFileWrite` not sufficient?
i.e. `request.ProjectStyle != ProjectStyle.ProjectJson` looks redundant. Am I missing something here?